pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -19,8 +19,27 @@ type EncodeRequest struct {
 	TestOn       string
 }
 
+type codecTestcase[T any] struct {
```

Comment:
- defined CodecTester so we use the `Setup(t T)` without the `startCR` flag for the `codec`
- defined `codecTestcase` so we don't use the `startCR` testcase flag for `codec` testcases
- defined `runCodecTests` to execute `codec` specific tests using the `CodecTester` interface with `Setup(t T)`